### ISSUE TITLE
Connection via X509 not possible

### ISSUE THREAD

**Issue by PixaGitHub** [nightroman/Mdbc#73]
Hello together,

we are using the Mdbc package to connect to our MongoDB by using SCRAM authentication. To make this way more secured we wanted to switch to a certificate based x.509 authentication. Now the Mdbc package is not connecting anymore with the new connection string. The connetion via the MongoShell or MongoCompass with the same connection string is working fine.
My questions:
- Is the x509 authentication not supported by your package?
- Do we need to follow exactly this connection string format?  mongodb://[username:password@]hostname[:port][/[database][?options]]

KR
Philip

**Comment by nightroman** [comment]
I have just released Mdbc v6.6.2 with the updated C# driver 2.18.0.
I do not think it will help but try the latest module, just in case.

> Is the x509 authentication not supported by your package?

Honestly, I do not know. Mdbc does not support anything itself, Mdbc uses the
C# driver and exposes some of its basic methods in PowerShell friendly way.

I am not a MongoDB guru, especially remote. I use MongoDB 100% locally.

> Do we need to follow exactly this connection string format? mongodb://[username:password@]hostname[:port][/[database][?options]]

I think currently it's the only way, if you use `Connect-Mdbc`.

Below is the alternative possible way using `MongoClientSettings` and `MongoClient` directly.
Try it, in theory `MongoClientSettings` should give you the best control possible.
(NB I will eventually add this way to `Connect-Mdbc`.)

```powershell
Import-Module Mdbc

### Use C# driver MongoClientSettings, "inconvenient" but full control

$set = [MongoDB.Driver.MongoClientSettings]::new()
$set.Server = [MongoDB.Driver.MongoServerAddress]::new('localhost', 27017)

# show available settings
#$set | Format-List | Out-String

# connect and get database "manually"
$client = [MongoDB.Driver.MongoClient]::new($set)
$database = $client.GetDatabase('test')

### Now use Mdbc

# get collection by Mdbc
$collection = Get-MdbcCollection test -NewCollection

# add and get data by Mdbd
@{_id=1; name='John'} | Add-MdbcData
Get-MdbcData
```

**Comment by nightroman** [comment]
Any progress and luck with the suggested way? It looks promising. Here is API docs
https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/T_MongoDB_Driver_MongoClientSettings.htm

**Comment by nightroman** [comment]
Another possible way is using `MongoUrl` and related `MongoUrlBuilder` for creating the client, similar to using settings.
https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/T_MongoDB_Driver_MongoUrl.htm
https://mongodb.github.io/mongo-csharp-driver/2.18/apidocs/html/T_MongoDB_Driver_MongoUrlBuilder.htm

**Comment by nightroman** [comment]
I have just released Mdbc v6.6.3 with the new parameters `Settings` and `Url` added to `Connect-Mdbc`. This should make connecting the client easier with all options available. I think this is all that Mdbc designed to do. I'm closing the issue.
